### PR TITLE
Bug fixes and automatic parameter iam policies

### PR DESCRIPTION
**Pull request by jvisker**
This allows fargate to receive the IAM permissions it needs to use the parameters that are requested. It also makes fixes to facilitate a v1 version of thsi module.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -78,49 +89,64 @@ module "alb" {
...
   ]
-  container_image         = module.ecr_initial_image.ecr_image_url
+  task_cpu                = var.task_cpu
+  task_memory             = var.task_memory
+  container_image         = var.container_url
   container_env_variables = var.container_env_variables
   container_secrets       = var.container_secrets
-  task_policies           = var.task_policies
-
+  task_policies           = length(aws_iam_policy.secrets_access) > 0 ? [aws_iam_policy.secrets_access[0].arn] : []
```
Comment (joshgubler) [inline main.tf]: merge var.task_policies

File: variables.tf [hunk 2]
```diff
@@ -48,4 +45,36 @@ variable "tags" {
   type = map(string)
   description = "A map of AWS Tags to attach to each resource created"
   default = {}
+}
+variable "container_url" {
+  type = string
```
Comment (joshgubler) [inline variables.tf]: add description

File: variables.tf [hunk 3]
```diff
@@ -48,4 +45,36 @@ variable "tags" {
...
+  default     = 512
+}
+variable "security_groups" {
+  type = list(string)
+  description = "Extra security groups"
+  default = []
+}
+variable "vpn_to_campus" {
+  type = bool
+  description = "VPN to Campus"
```
Comment (joshgubler) [inline variables.tf]: "Do the Fargate tasks need to run in the VPC that has a VPN back to campus?"

File: variables.tf [hunk 4]
```diff
@@ -48,4 +45,36 @@ variable "tags" {
...
+  default = false
+}
+variable "log_retention_in_days" {
+  type = number
+  default = 7
```
Comment (joshgubler) [inline variables.tf]: add a description

File: variables.tf [hunk 5]
```diff
@@ -48,4 +45,36 @@ variable "tags" {
   type = map(string)
   description = "A map of AWS Tags to attach to each resource created"
   default = {}
+}
+variable "container_url" {
```
Comment (joshgubler) [inline variables.tf]: maybe "container_image_url"

File: variables.tf [hunk 6]
```diff
@@ -48,4 +45,36 @@ variable "tags" {
...
+  default = false
+}
+variable "log_retention_in_days" {
+  type = number
+  default = 7
+}
+variable "health_check_grace_period" {
+  type        = number
+  description = "Health check grace period in seconds. Defaults to 0."
+  default     = 0
```
Comment (joshgubler) [inline variables.tf:81]: If we're copying defaults from lower modules...I think you can set the default to `null` and Terraform will act like you didn't set the line at all, and the lower module's default will be used.
I don't know if this is better or not, but it's an option.

Comment (jvisker) [reply variables.tf:81]: If I set the default as null it tries to set the value as null

File: main.tf [hunk 7]
```diff
@@ -78,49 +89,64 @@ module "alb" {
...
-  container_image         = module.ecr_initial_image.ecr_image_url
+  task_cpu                = var.task_cpu
+  task_memory             = var.task_memory
+  container_image         = var.container_image_url
   container_env_variables = var.container_env_variables
   container_secrets       = var.container_secrets
-  task_policies           = var.task_policies
-
+  task_policies           = concat(length(aws_iam_policy.secrets_access) > 0 ? [aws_iam_policy.secrets_access[0].arn] : [], var.task_policies)
+  task_execution_policies = length(aws_iam_policy.secrets_access) > 0 ? [aws_iam_policy.secrets_access[0].arn] : []
```
Comment (yoshutch) [inline main.tf:134]: can you use `local.has_secrets` in your conditional statements here?

File: main.tf [hunk 8]
```diff
@@ -78,49 +89,64 @@ module "alb" {
...
+    blue_target_group_name                         = lookup(module.alb.target_groups, "blue", null) != null ? module.alb.target_groups["blue"].name : null
+    green_target_group_name                        = lookup(module.alb.target_groups, "green", null) != null ? module.alb.target_groups["green"].name : null
     service_role_arn                               = module.acs.power_builder_role.arn
   }
 
   module_depends_on             = [module.alb.alb]
   role_permissions_boundary_arn = module.acs.role_permissions_boundary.arn
-
+  log_retention_in_days = var.log_retention_in_days
+  health_check_grace_period = var.health_check_grace_period
```
Comment (yoshutch) [inline main.tf]: (nit) run `terraform fmt`

File: main.tf (hunk 7 above)
Comment (jvisker) [reply main.tf:134]: It was to solve a weird problem I got doing a destroy

**Review by joshgubler — APPROVED** [review]

**Pull request merged**